packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -0,0 +1,386 @@
+			$additional_fields = "
+				{$this->range_field} as range_index,
+			    {$key_fields},
+			";
+		}
+
+		$filter_stamenet = $this->build_filter_statement( $range_from, $range_to, $filter_values );
+
+		$query = "
+			SELECT
+				{$additional_fields}
+				SUM(
+					CRC32(
+						CONCAT_WS( '#', {$salt}, {$checksum_fields_string} )
+					)
+				)  AS checksum
+			 FROM
+			    {$this->table}
+			 WHERE
+				{$filter_stamenet}
+		";
+
+		/**
+		 * We need the GROUP BY only for compound keys
+		 */
+		if ( $granular_result ) {
+			$query .= "
+				GROUP BY {$key_fields}
+			";
+		}
+
+		return $query;
+
+	}
+
+	public function get_range_edges( $range_from = null, $range_to = null, $limit = null ) {
+		global $wpdb;
+
+		$this->validate_fields( array( $this->range_field ) );
+
+		// `trim()` to make sure we don't add the statement if it's empty
+		$filters = trim( $this->build_filter_statement( $range_from, $range_to ) );
+
+		$filter_statement = '';
+		if ( ! empty( $filters ) ) {
+			$filter_statement = "
+				WHERE
+					{$filters}
+			";
+		}
+
+		// Only make the distinct count when we know there can be multiple entries for the range column
+		$distinct_count = count( $this->key_fields ) > 1 ? 'DISTINCT' : '';
+
+		$query = "
+			SELECT
+			       MIN({$this->range_field}) as min_range,
+			       MAX({$this->range_field}) as max_range,
+			       COUNT( {$distinct_count} {$this->range_field}) as item_count
+			FROM
+		";
+
+		/**
+		 * If `$limit` is not specified, we can directly use the table
+		 */
+		if ( ! $limit ) {
+			$query .= "
+				{$this->table}
+	            {$filter_statement}
+			";
+		} else {
+			/**
+			 * If there is `$limit` specified, we can't directly use `MIN/MAX()` as they don't work with `LIMIT`.
+			 * That's why we will alter the query for this case.
+			 */
+			$limit = intval( $limit );
+
+			$query .= "
+				(
+					SELECT
+						{$distinct_count} {$this->range_field}
+					FROM
+						{$this->table}
+						{$filter_statement}
+					ORDER BY
+						{$this->range_field} ASC
+					LIMIT {$limit}
+				) as ids_query
+			";
+		}
+
+		$result = $wpdb->get_row( $query, ARRAY_A );
+
+		if ( ! $result || ! is_array( $result ) ) {
+			throw new Exception( 'Unable to get range edges' );
+		}
+
+		return $result;
+	}
+
+	public function prepare_results_for_output( &$results ) {
+		// get the compound key
+		// only return range and compound key for granular results
+
+		foreach ( $results as &$result ) {
+			// Working on reference to save memory here.
+
+			$key = array();
+			foreach ( $this->key_fields as $field ) {
+				$key[] = $result[ $field ];
+			}
+
+			$result = array(
+				'key'      => implode( '-', $key ),
+				'checksum' => $result['checksum'],
+			);
+
+		}
+	}
+
+	public function calculate_checksum( $range_from = null, $range_to = null, $filter_values = null, $granular_result = false, $simple_return_value = true ) {
```

Comment:
we need a check on the range_from and range_to

we should really not allow a -1 type query . instead it should default to a reasonable value 1,000 ? 10,000?
also validate range_to should be > range_from

Reply:
The worst thing that will happen is that there will be no results returned. Capping it is a good thing, but sometimes we might want to run over the cap that we set in code. 

We might add a large-enough number to prevent any madness, but not supplying it would be the same as going over the limit.

Do you think we need to have a upper/lower limit performance-wise?